ontotext-graphql-playground-component/src/mappers/fetcher-configuration.mapper.ts

```diff
@@ -0,0 +1,26 @@
+import {ExternalGraphqlPlaygroundConfiguration} from '../models/external-graphql-playground-configuration';
+import {Fetcher} from '../graphiql-models/fetcher';
+
+/**
+ * Utility class for converting an external GraphQL playground configuration into a fetcher configuration compatible with GraphiQL.
+ */
+export class FetcherConfiguration {
+  
+  /**
+   * Converts an external GraphQL playground configuration into a fetcher configuration.
+   *
+   * @param {ExternalGraphqlPlaygroundConfiguration} configuration - The external configuration containing the GraphQL endpoint and optional headers.
+   * @returns {object} A fetcher configuration object containing the endpoint URL and headers.
+   */
+  static toFetcherConfiguration(configuration: ExternalGraphqlPlaygroundConfiguration): Fetcher {
+    const fetcherConfig = {
+      url: configuration.endpoint,
+    }
+    // If headers are provided, add them to the fetcher configuration. We currently just pass them as they are without
+    // any validation.
+    if (configuration.headers) {
+      fetcherConfig['headers'] = configuration.headers;
+    }
+    return fetcherConfig;
```

Comment:
I am confused. Why the Fetcher type is defined like this `export type Fetcher = {}` if we put other stuff inside. Improve the type definition!

Reply:
It has been removed, and now the types provided by GraphiQL are used.